.travis.yml

```diff
@@ -0,0 +1,85 @@
+sudo: required
+dist: trusty
+language: cpp
+compiler:
+    # FIXME: For now, building with Clang is disabled because the STP built with
+    # it hits an assertion failure during some tests. We should sort this out
+    # eventually and file the bug against STP or Clang whichever is appropriate,
+    # but for now it is easier to just reduce the number of configs we test
+    # anyway.
+    # - clang
+
+    - gcc
+env:
+    ###########################################################################
+    # Configurations
+    #
+    # Each line in the "env" section represents a set of environment variables
+    # passed to a build. Thus each line represents a different build
+    # configuration.
+    ###########################################################################
+
+    # Check a subset of the matrix of:
+    #   LLVM  : {2.9, 3.4}
+    #   SOLVERS : {STP:Z3}
+    #   STP_VERSION : {2.1.2}
+    #   DISABLE_ASSERTIONS: {0}
+    #   ENABLE_OPTIMIZED: {1}
+    #   USE_CMAKE: {0, 1}
+
+    matrix:
+    # Check KLEE CMake build in a few configurations
+    - LLVM_VERSION=3.4 SOLVERS=STP:Z3 STP_VERSION=2.1.2 DISABLE_ASSERTIONS=0 ENABLE_OPTIMIZED=1 USE_CMAKE=1
+    - LLVM_VERSION=2.9 SOLVERS=STP:Z3 STP_VERSION=2.1.2 DISABLE_ASSERTIONS=0 ENABLE_OPTIMIZED=1 USE_CMAKE=1
+
+    # Check KLEE Autoconf build in a few configurations
+    - LLVM_VERSION=3.4 SOLVERS=STP:Z3 STP_VERSION=2.1.2 DISABLE_ASSERTIONS=0 ENABLE_OPTIMIZED=1 USE_CMAKE=0
+    - LLVM_VERSION=2.9 SOLVERS=STP:Z3 STP_VERSION=2.1.2 DISABLE_ASSERTIONS=0 ENABLE_OPTIMIZED=1 USE_CMAKE=0
+
+    global:
```

Comment:
Please rip out these `secure`variables. Those are for coverage which we aren't doing here. If you change that the `global` key can be removed and we could also drop `matrix` and just have the list live underneath `env` like in https://docs.travis-ci.com/user/environment-variables#Defining-public-variables-in-.travis.yml